TMQTTClient/MQTT.pas

```diff
@@ -266,7 +266,7 @@       TMQTTClient = class(TObject)
           if SocketWrite(Data) then
             begin
               Result := True;
-              FReadThread.Terminate;
+              FReadThread.waitFor;
```

Comment:
if the server did not close the socket (lost the package or the server application has hung), then this code will cause the client application to hang.